adapters/httpapi_winhttp.c

```diff
@@ -476,7 +479,20 @@ static HTTPAPI_RESULT InitiateWinhttpRequest(HTTP_HANDLE_DATA* handleData, HTTPA
     ))
     {
         result = HTTPAPI_SET_X509_FAILURE;
-        LogErrorWinHTTPWithGetLastErrorAsString("unable to WinHttpSetOption");
+        LogErrorWinHTTPWithGetLastErrorAsString("unable to WinHttpSetOption (WINHTTP_OPTION_CLIENT_CERT_CONTEXT)");
+        (void)WinHttpCloseHandle(*requestHandle);
+        *requestHandle = NULL;
+    }
+    else if ((handleData->x509SchannelHandle != NULL) &&
+        !WinHttpSetOption(
+            *requestHandle,
+            WINHTTP_OPTION_SECURE_PROTOCOLS,
+            &supported_protocols,
+            sizeof(supported_protocols)
+        ))
+    {
+        result = HTTPAPI_SET_X509_FAILURE;
```

Comment:
Recommend different, more generic error code?

Reply:
Went with a different/specific error code.

---
In reply to: [331298895](https://github.com/Azure/azure-c-shared-utility/pull/391#discussion_r331298895) [](ancestors = 331298895)